src/box/box.cc

```diff
@@ -1375,10 +1375,22 @@ box_process1(struct request *request, box_tuple_t **result)
 	struct space *space = space_cache_find(request->space_id);
 	if (space == NULL)
 		return -1;
+	struct engine *engine = (struct engine *)space->engine;
 	if (!space_is_temporary(space) &&
 	    space_group_id(space) != GROUP_LOCAL &&
 	    box_check_writable() != 0)
 		return -1;
+
+	if(engine && !strcmp(engine->name, "memtx")) {
```

Comment:
It's better to check `engine->id == 0` to determine whether it's memtx, see `space_is_memtx` function. I guess it would be faster.

Reply:
ok